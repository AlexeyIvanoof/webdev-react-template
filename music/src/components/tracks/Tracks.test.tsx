import React from 'react';
import { render } from '@testing-library/react';
//import '@testing-library/jest-dom/extend-expect'; // для дополнительных совпадений
import Tracks from './Track';
import ReduxProvider from '@/store/ReduxProvider';


describe('Tracks Component', () => {
  it('renders correctly', () => {
    const tracks = [
      { _id: 1, title: 'Track 1', author: 'Artist 1', album: 'Album 1', duration_in_seconds: 225 },
      { _id: 2, title: 'Track 2', author: 'Artist 2', album: 'Album 2', duration_in_seconds: 260 },
    ];

    const title = 'My Playlist';

    const { asFragment } = render(
        <ReduxProvider>
                 <Tracks filteredTracks={tracks} tracks={tracks} title={title} />
        </ReduxProvider>
);

    // снимок визуализированного компонента
    expect(asFragment()).toMatchSnapshot();
  });
});
