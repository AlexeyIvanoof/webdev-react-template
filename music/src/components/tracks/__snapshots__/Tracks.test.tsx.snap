// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tracks Component renders correctly 1`] = `
<DocumentFragment>
  <h2
    class="centerblockH2"
  >
    My Playlist
  </h2>
  <div
    class="centerblockFilter"
  >
    <div
      class="filterTitle"
    >
      Искать по:
    </div>
    <div>
      <div
        class="filterBlock"
      >
        <div
          class="filterButton btnText"
        >
          author
        </div>
      </div>
    </div>
    <div>
      <div
        class="filterBlock"
      >
        <div
          class="filterButton btnText"
        >
          genre
        </div>
      </div>
    </div>
    <div>
      <div
        class="filterBlock"
      >
        <div
          class="filterButton btnText"
        >
          году выпуска
        </div>
      </div>
    </div>
  </div>
  <div
    class="centerblockContent"
  >
    <div
      class="contentTitle"
    >
      <div
        class="playlistTitleCol col01"
      >
        Трек
      </div>
      <div
        class="playlistTitleCol col02"
      >
        Исполнитель
      </div>
      <div
        class="playlistTitleCol col03"
      >
        Альбом
      </div>
      <div
        class="playlistTitleCol col04"
      >
        <svg
          class="playlistTitleSvg"
        >
          <use
            xlink:href="/sprite.svg#icon-watch"
          />
        </svg>
      </div>
    </div>
    <div
      class="contentPlaylist"
    >
      <div
        class="playlistItem"
      >
        <div
          class="playlistTrack"
        >
          <div
            class="trackTitle"
          >
            <div
              class="trackTitleImage"
            >
              <svg
                class="trackTitleSvg"
              >
                <use
                  xlink:href="/sprite.svg#icon-note"
                />
              </svg>
            </div>
            <div>
              <a
                class="trackTitleLink"
              >
                 
                <span
                  class="trackTitleSpan"
                />
              </a>
            </div>
          </div>
          <div
            class="trackAuthor"
          >
            <a
              class="trackAuthorLink"
            >
              Artist 1
            </a>
          </div>
          <div
            class="trackAlbum"
          >
            <a
              class="trackAlbumLink"
            >
              Album 1
            </a>
          </div>
          <div>
            <svg
              class="trackTimeSvg"
            >
              <use
                xlink:href="/sprite.svg#icon-like"
              />
            </svg>
            <span
              class="trackTimeText"
            >
              03:45
            </span>
          </div>
        </div>
      </div>
      <div
        class="playlistItem"
      >
        <div
          class="playlistTrack"
        >
          <div
            class="trackTitle"
          >
            <div
              class="trackTitleImage"
            >
              <svg
                class="trackTitleSvg"
              >
                <use
                  xlink:href="/sprite.svg#icon-note"
                />
              </svg>
            </div>
            <div>
              <a
                class="trackTitleLink"
              >
                 
                <span
                  class="trackTitleSpan"
                />
              </a>
            </div>
          </div>
          <div
            class="trackAuthor"
          >
            <a
              class="trackAuthorLink"
            >
              Artist 2
            </a>
          </div>
          <div
            class="trackAlbum"
          >
            <a
              class="trackAlbumLink"
            >
              Album 2
            </a>
          </div>
          <div>
            <svg
              class="trackTimeSvg"
            >
              <use
                xlink:href="/sprite.svg#icon-like"
              />
            </svg>
            <span
              class="trackTimeText"
            >
              04:20
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
